### ISSUE TITLE
Linux controllers no longer detected since 1.4.8

### ISSUE THREAD

**Issue by JustAPotota** [defold/defold#8059]
**Describe the bug (REQUIRED)**
I tried to use my wired Xbox One controller on 1.6.0, but it wasn't detected by my project or GDC. Checking older versions of GDC, it only works on versions <1.4.8.

**To Reproduce (REQUIRED)**
1. Connect controller
2. Run GDC (or any Defold project)

**Expected behavior (REQUIRED)**
It should be detected, even if unmapped.

**Defold version (REQUIRED):**
1.6.0

**Platforms (REQUIRED):**
Fedora Silverblue 39
![image](https://github.com/defold/defold/assets/43710172/1f954537-d794-4104-bfe7-d8e680691b96)

**Logs (OPTIONAL):**
```
[Downloads]$ ./gdc-linux147
INFO:GRAPHICS: Initialised graphics device 'ADAPTER_TYPE_OPENGL'
WARNING:HID: There was no callback function set to handle the gamepad connection!
WARNING:HID: There was no callback function set to handle the gamepad connection!
1: Corsair CORSAIR K70 RGB MK.2 Mechanical Gaming Keyboard Mouse
2: PowerA Xbox One wired controller

* Which gamepad do you want to calibrate? [1-2] ^C

[Downloads]$ ./gdc-linux148
INFO:GRAPHICS: Initialised graphics device 'ADAPTER_TYPE_OPENGL'
No connected gamepads, rechecking in 3 seconds. 2.. 1..

[Downloads]$ ./gdc-linux160
INFO:GRAPHICS: Initialised graphics device 'ADAPTER_TYPE_OPENGL'
No connected gamepads, rechecking in 3 seconds. 2.. 1..
```

**Comment by Jhonnyg** [comment]
Do you have access to any other controllers?

**Comment by JustAPotota** [comment]
I tested a Gamecube controller via a third-party adapter, same result as the Xbox controller:
```
[Downloads]$ ./gdc-linux148
INFO:GRAPHICS: Initialised graphics device 'ADAPTER_TYPE_OPENGL'
No connected gamepads, rechecking in 3 seconds. 2.. 1.. Rechecking...

[Downloads]$ ./gdc-linux147
INFO:GRAPHICS: Initialised graphics device 'ADAPTER_TYPE_OPENGL'
WARNING:HID: There was no callback function set to handle the gamepad connection!
WARNING:HID: There was no callback function set to handle the gamepad connection!
WARNING:HID: There was no callback function set to handle the gamepad connection!
WARNING:HID: There was no callback function set to handle the gamepad connection!
WARNING:HID: There was no callback function set to handle the gamepad connection!
1: Corsair CORSAIR K70 RGB MK.2 Mechanical Gaming Keyboard Mouse
2: ShanWan     WUP-028
3: ShanWan     WUP-028
4: ShanWan     WUP-028
5: ShanWan     WUP-028

* Which gamepad do you want to calibrate? [1-5] ^C
```
Creating bindings with GDC works just fine for both controllers on <1.4.8. I also have a Switch Pro Controller I could try, but it would probably give the same result.

**Comment by 8bitskull** [comment]
I have tested on the Steam Deck (which runs on Linux) and can reproduce this issue.

1.4.7 - gamepad detected (as "Microsoft X-Box 360 pad 0"
1.4.8 - no gamepad detected

@Jhonnyg let me know if I can assist in any way! This issue basically means Defold games don't really work on Steam Deck anymore so I'm keen to help get it resolved.

**Comment by JCash** [comment]
The commits between those versions (`git log 1.4.7..1.4.8 --oneline`):
```bash
fa5980d7a1 (tag: 1.4.8) Fix markdown viewer
eee6afb3af Fix issue when `arm64-macos` can't be bundle without bundling `x86_64-macos` (#7792)
081c6953db Fix compile exceptions regression (#7786)
931d2ab65b Fix for lua panic when connecting a gamepad (#7784)
5ab36f5e26 Fix for reboot issue in the HID system (#7777)
928fda7dbb Custom vertex formats for sprites (#7508)
1bacda0b22 Merge branch 'dev' of github.com:defold/defold into dev
8f5e836c5e Bundle only specified architectures in HTML5 builds (#7544)
cd76241a73 (tag: 1.4.8-beta) Merge branch 'editor-dev' into dev
96b00c8324 Merge branch 'editor-dev' of github.com:defold/defold into editor-dev
401f1a4405 Properly log engine build errors (#7773)
a87ca67c75 Update JavaFX (#7770)
a1e7d612bc Fix running tests on mac (#7768)
a64bce5543 FIxed issue of LuaJIT not working correctly on arm64-macos (#7764)
03febf3189 Upgrade to OpenJDK 17 (#7450)
fce2e2a425 Individual materials for GUI nodes (#7606)
802008a3c7 Particle effects crashfix (#7766)
bd0145535d Add view, projection and aspect_ratio as camera properties (#7762)
69f60fc371 Add gui.get_tree() (#7747)
0e137150d7 Check node existence before querying it on load (#7765)
86d062d55e Build remote engine and tasks in parallel (#7757)
a265a1714d Support multiple materials for model component (#7380)
76ad15e54a Make sure to call the animation callback (#7760)
2b011e7497 Implement lazy loading of the debugged app state (#7744)
9aa74c652a (tag: 1.4.8-alpha) Bumped version to 1.4.8
9903c9cf48 Merge branch 'editor-dev' into dev
94b853c671 Merge branch 'master' into editor-dev
b3e46f0956 Expose version information to editor extensions (#7746)
95ebfe996a Merge branch 'master' into editor-dev
53ddb10620 Show model's Vertex Count in profiler for models in Local Vector Space (#7748)
f2f7f12975 Typo for Win32 header when used on Linux
c1b9704f04 Refactored code from the console branch (#7733)
a3233321d9 Determine unmapped gamepad at runtime (#7740)
64af1dc38c Added the used vector math functions to dmVMath namespace in dmSdk (#7732)
e92ab7ed4b fix issue with debugger (#7735)
de77e1c99f Fix broken link in COMPLYING_WITH_LICENSES.md (#7731)
a8799a0169 Added PSSL enum in order to be able to use vanilla bob for building PS projects
734d372d88 Merge branch 'dev' of github.com:defold/defold into dev
4c68b294f4 FIxed issue of builtin shader content not building for win32
905f8529a4 Added support for unpacking .dmanifest, .modelc, .fpc, .vpc and .luac (#7725)
da91d5e7dd Add error output when the rest api fails (e.g. when token has missing premissions) (#7726)
7ed48414f7 Fix for building builtins with correct platform
24ed6ac132 Merge remote-tracking branch 'origin/editor-dev' into dev
bd9d5a1025 Migrate to our managed sass dependency (#7728)
5e87313c74 Replace amazonica with aws-api library (#7721)
3bb3be870b Added PS5 platform enum (#7723)
```

looking at that list, a few commits stand out:
```
931d2ab65b Fix for lua panic when connecting a gamepad (#7784)
a3233321d9 Determine unmapped gamepad at runtime (#7740)
c1b9704f04 Refactored code from the console branch (#7733) (refactoring in engine/input and default.gamepads)
```

I havent tried this code using a Linux VM, so I'm not sure how to best test this yet.
